src/librbd/io/Utils.h

```diff
@@ -34,6 +34,11 @@ void read_parent(ImageCtxT *image_ctx, uint64_t object_no, const Extents &extent
                  librados::snap_t snap_id, const ZTracer::Trace &trace,
                  ceph::bufferlist* data, Context* on_finish);
 
+template <typename ImageCtxT = librbd::ImageCtx>
+int clip_request(ImageCtxT *image_ctx, io::Extents &image_extents);
```

Comment:
Nit: prefer pointer for in-out params

Reply:
Do you mean to change io::Extents& to io::Extents *?
```cpp
template <typename ImageCtxT = librbd::ImageCtx>
int clip_request(ImageCtxT *image_ctx, io::Extents *image_extents);
```

Reply:
Correct -- we try to use pointers for in-out parameters so that there is no hidden expectations of data being changed.

Reply:
Thank you. Updated.